changing to zap log

nithu0115 opened the pull request:
Issue #, if available: N/A

Description of changes:

Switching to zap core logging instead of `klog`. Part of the motivation for `logr` was the idea that V level is a bad idea - log lines are all INFO or ERROR. But there are real world scenario when we want know more on what is happening and it is required. We should limit to V(1) (Debug logging) and should not go over that level. 

Changes:
1) Added log-level arg which will either take `debug` or `info` log levels 
2) Modified manager.go template to pass in region value to generate create temp credentials to be consistent with config region flag





By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

nithu0115 commented:
(Note to reviewer: Change to templates/pkg/crd_manager.go.tpl will require us to re-generate Service controllers)

A-Hilaly reviewed (approved):
/lgtm 👍 

Shall we regenerate the service controllers in this PR ?

A-Hilaly commented on pkg/runtime/config.go:
```diff
@@ -28,6 +29,7 @@ const (
 	flagEnableDevLogging     = "enable-development-logging"
 	flagAWSAccountID         = "aws-account-id"
 	flagAWSRegion            = "aws-region"
+	flagLogLevel			 = "log-level"
```
nit: please run go fmt :)

A-Hilaly commented on pkg/runtime/session.go:
```diff
@@ -13,12 +13,16 @@
 
-import "github.com/aws/aws-sdk-go/aws/session"
+import (
+	ackv1alpha1 "github.com/aws/aws-controllers-k8s/apis/core/v1alpha1"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
```
nit: let's keep internal and external imports separated

A-Hilaly commented on pkg/runtime/reconciler.go:
```diff
@@ -15,7 +15,6 @@ package runtime
 
 import (
 	"context"
-
```
the new line was used to keep standard imports and external imports separated

A-Hilaly reviewed (commented):
leaving few comments inline

nithu0115 commented:
> Shall we regenerate the service controllers in this PR ?

I am happy to re-generate but wanted to check in with @jaypipes to see if we want to introduce any process around this. Fwiw, we should maintain 2 branches (main and dev/staging). This way, whenever we make such breaking changes, we can push to dev and cherry pick to merge into main once we cut new branches.

nithu0115 commented:
> > Shall we regenerate the service controllers in this PR ?
> 
> I am happy to re-generate but wanted to check in with @jaypipes to see if we want to introduce any process around this. Fwiw, we should maintain 2 branches (main and dev/staging). This way, whenever we make such breaking changes, we can push to dev and cherry pick to merge into main once we cut new releases.

Spoke to @jaypipes offline and come to a consensus to re-generate service controllers as this is a minimal change. If the change is breaking one or a big one, we will go ahead and merge the PR first without re-generating service controller(s) and then open new PRs for each controller.

jaypipes commented on pkg/runtime/reconciler.go:
```diff
@@ -98,11 +98,12 @@ func (r *reconciler) reconcile(req ctrlrt.Request) error {
 		"region", region,
 		"kind", r.rd.GroupKind().String(),
 	)
+	log.Info("starting reconcilation")
```
This should be DEBUG, not INFO :)

nithu0115 replied:
Done :)

jaypipes commented on pkg/runtime/session.go:
```diff
@@ -13,12 +13,17 @@
 	// NOTE(jaypipes): session.NewSession() is needed for the STS::AssumeRole
 	// stuff we will need to do...
-	sess, err := session.NewSession()
+	sess, err := session.NewSession(aws.NewConfig().WithRegion(string(region)))
```
This change is not related to logging. Would you mind removing this change (put it into a separate PR please)

jaypipes commented on services/bookstore/pkg/resource/book/manager.go:
```diff
@@ -150,7 +150,7 @@ func newResourceManager(
 	region ackv1alpha1.AWSRegion,
 ) (*resourceManager, error) {
-	sess, err := ackrt.NewSession()
+	sess, err := ackrt.NewSession(region)
```
ditto.

jaypipes commented on services/petstore/pkg/resource/pet/manager.go:
```diff
@@ -227,7 +227,7 @@ func newResourceManager(
 	region ackv1alpha1.AWSRegion,
 ) (*resourceManager, error) {
-	sess, err := ackrt.NewSession()
+	sess, err := ackrt.NewSession(region)
```
ditto

jaypipes commented on services/ecr/pkg/resource/repository/manager.go:
```diff
@@ -153,7 +153,7 @@ func newResourceManager(
 	region ackv1alpha1.AWSRegion,
 ) (*resourceManager, error) {
-	sess, err := ackrt.NewSession()
+	sess, err := ackrt.NewSession(region)
```
ditto.

jaypipes commented on services/sns/config/controller/deployment.yaml:
```diff
@@ -27,11 +27,11 @@ spec:
-        - "$(AWS_REGION)"
+        - "${AWS_REGION}"
         - --enable-development-logging
-        - "$(ACK_ENABLE_DEVELOPMENT_LOGGING)"
```
These changes need to be undone. The parentheses are necessary. The curly braces are incorrect for Docker args.

jaypipes commented on templates/config/controller/deployment.yaml.tpl:
```diff
@@ -27,11 +27,11 @@ spec:
+        - "${AWS_REGION}"
         - --enable-development-logging
-        - "$(ACK_ENABLE_DEVELOPMENT_LOGGING)"
+        - "${ACK_ENABLE_DEVELOPMENT_LOGGING}"
```
These changes need to be undone. The parentheses are necessary. The curly braces are incorrect for Docker args.

nithu0115 replied:
I wanted to keep this consistent with what we currently have - https://github.com/aws/aws-controllers-k8s/blob/main/services/ecr/config/controller/deployment.yaml#L34. Thus, I made this change.

jaypipes reviewed (changes requested):
@nithu0115 this needs to be redone. I support the changes for logging, but the remaining changes either belong in a separate PR (NewSession(region)) or are incorrect (the curly braces instead of parens)

nithu0115 commented:
> @nithu0115 this needs to be redone. I support the changes for logging, but the remaining changes either belong in a separate PR (NewSession(region)) or are incorrect (the curly braces instead of parens)

Addressed your comments.

jaypipes reviewed (approved):
:+1:
